app/controllers/main_controller.rb

```diff
@@ -60,7 +60,7 @@ def login
     session[:redirect_uri] = nil
     refresh_timeout
     # redirect to last visited page or to main page
-    redirect_to(uri || { controller: 'courses', action: 'index' })
+    found_user.admin_user? ? redirect_to(admin_path) : redirect_to(uri || { controller: 'courses', action: 'index' })
```

Comment:
The ordering is incorrect here.

First, if `uri.present?` is true then the user should be redirected to it. Otherwise redirect to either the `admin_path` or `courses_path` (you can change the hash syntax to the path helper syntax). Don't try to do this in one line, it's a good candidate for an if statement.